// components/ImageGroup/ImageGroup.js
import React, { Component } from 'react'
import PropTypes from 'prop-types'

let images4 = ['https://images7.alphacoders.com/411/thumb-1920-411820.jpg',
  'https://images.pexels.com/photos/33045/lion-wild-africa-african.jpg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
  'https://images.pexels.com/photos/753626/pexels-photo-753626.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260',
  'https://images.pexels.com/photos/490466/pexels-photo-490466.jpeg?auto=compress&cs=tinysrgb&dpr=2&h=750&w=1260']

let data

class ImageGroup extends Component {

  constructor(props) {
    super(props)
    this.state = {
      data: props.data,
      width: props.width,
      height: props.height,
      borderRad: props.borderRad,
      activeImage: 'asd',
      activeImageIndex: 0,
    }

    this.imageClicked = this.imageClicked.bind(this)

  }

  componentWillReceiveProps(nextProps) {

    if (nextProps.data !== data) {
      data = nextProps.data
    }

  }

  imageClicked(url, index) {
    if (this.props.imageClicked !== undefined) {
      this.props.imageClicked(url, index, this.state.data)
    }
  }

  renderOne() {

    let str1 = this.state.data[0]

    let style = {
      width: '100%', height: '100%',
      backgroundImage: `url(${str1})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    return (<div style={style} onClick={() => this.imageClicked(str1, 0)}></div>)

  }

  renderTwo() {

    let str1 = this.state.data[0]
    let str2 = this.state.data[1]

    let style1 = {
      width: '50%', height: '100%',
      backgroundImage: `url(${str1})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    let style2 = {
      width: '50%', height: '100%',
      backgroundImage: `url(${str2})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    return (
      <div style={{ width: '100%', height: '100%', display: 'flex', flexDirection: 'row' }}>
        <div style={style1} onClick={() => this.imageClicked(str1, 0)}></div>
        <div style={style2} onClick={() => this.imageClicked(str2, 1)}></div>
      </div>
    )

  }

  renderThree() {

    let str1 = this.state.data[0]
    let str2 = this.state.data[1]
    let str3 = this.state.data[2]

    let style1 = {
      width: '50%', height: '100%',
      backgroundImage: `url(${str1})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    let style2 = {
      width: '100%', height: '50%',
      backgroundImage: `url(${str2})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    let style3 = {
      width: '100%', height: '50%',
      backgroundImage: `url(${str3})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    return (
      <div style={{ width: '100%', height: '100%', display: 'flex', flexDirection: 'row' }}>
        <div style={style1} onClick={() => this.imageClicked(str1, 0)}></div>
        <div style={{ display: 'flex', flexDirection: 'column', width: '50%', height: '100%' }}>
          <div style={style2} onClick={() => this.imageClicked(str2, 1)}></div>
          <div style={style3} onClick={() => this.imageClicked(str3, 2)}></div>
        </div>
      </div>
    )

  }

  renderFour() {

    let str1 = this.state.data[0]
    let str2 = this.state.data[1]
    let str3 = this.state.data[2]
    let str4 = this.state.data[3]

    let style1 = {
      width: '100%', height: '50%',
      backgroundImage: `url(${str1})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    let style2 = {
      width: '100%', height: '50%',
      backgroundImage: `url(${str2})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    let style3 = {
      width: '100%', height: '50%',
      backgroundImage: `url(${str3})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    let style4 = {
      width: '100%', height: '50%',
      backgroundImage: `url(${str4})`,
      backgroundSize: 'cover',
      backgroundRepeat: 'no-repeat',
      cursor: 'pointer'
    }

    return (
      <div style={{ width: '100%', height: '100%', display: 'flex', flexDirection: 'row' }}>
        <div style={{ display: 'flex', flexDirection: 'column', width: '50%', height: '100%' }}>
          <div style={style1} onClick={() => this.imageClicked(str1, 0)}></div>
          <div style={style2} onClick={() => this.imageClicked(str2, 1)}></div>
        </div>
        <div style={{ display: 'flex', flexDirection: 'column', width: '50%', height: '100%' }}>
          <div style={style3} onClick={() => this.imageClicked(str3, 2)}></div>
          <div style={style4} onClick={() => this.imageClicked(str4, 3)}></div>
        </div>
      </div>
    )

  }

  render() {

    let length = this.state.data.length

    if (!length || length <= 0 || length > 4) {
      return ''
    }

    let mainStyle = {
      width: this.state.width, height: this.state.height,
      borderRadius: this.state.borderRad,
      overflow: 'hidden'
    }

    return (
      <div className="App" style={mainStyle}>
        {length === 1 ? this.renderOne() :
          length === 2 ? this.renderTwo() :
            length === 3 ? this.renderThree() :
              length === 4 ? this.renderFour() : ''}
      </div>
    )
  }

}

ImageGroup.propTypes = {
  width: PropTypes.string,
  height: PropTypes.string,
  borderRad: PropTypes.string,
  data: PropTypes.array,
  imageClicked: PropTypes.func,
}

ImageGroup.defaultProps = {
  width: '200px',
  height: '200px',
  borderRad: '0px',
  data: images4,
  imageClicked: undefined,
}

export default ImageGroup
